resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -256,7 +259,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   }
 
   override def sufficientResourcesRegistered(): Boolean = {
-    totalCoresAcquired >= maxCores * minRegisteredRatio
+    totalCoreCount.get >= maxCoresOption.getOrElse(0) * minRegisteredRatio
```

Comment:
This means return fast most of the time. `totalCoreCount` holds the total number of cores in the cluster. With a reasonable max core value this inequality will usually hold, right? This means immediate task schedule? In the Jira the opposite is described, am I missing something?